python/ray/ml/train/integrations/torch/torch_trainer.py

```diff
@@ -1,53 +1,169 @@
+    the "train" key), then it will be split into multiple dataset
+    shards that can then be accessed by ``ray.train.get_dataset_shard("train")`` inside
+    ``train_loop_per_worker``. All the other datasets will not be split and
+    ``ray.train.get_dataset_shard(...)`` will return the the entire Dataset.
+
+    Inside the ``train_loop_per_worker`` function, you can use any of the
+    :ref:`Ray Train function utils <train-api-func-utils>`.
+
+    .. code-block:: python
+
+        def train_loop_per_worker():
+            # Report intermediate results for callbacks or logging.
+            train.report(...)
+
+            # Checkpoints the provided args as restorable state.
+            train.save_checkpoint(...)
+
+            # Returns dict of last saved checkpoint.
+            train.load_checkpoint()
+
+            # Returns the Ray Dataset shard for the given key.
+            train.get_dataset_shard("my_dataset")
+
+            # Returns the total number of workers executing training.
+            train.get_world_size()
+
+            # Returns the rank of this worker.
+            train.get_world_rank()
+
+            # Returns the rank of the worker on the current node.
+            train.get_local_rank()
+
+    You can also use any of the :ref:`Torch specific function utils
+    <train-api-torch-utils>`.
+
+    .. code-block:: python
+        def train_loop_per_worker():
+            # Prepares model for distribted training by wrapping in
+            # `DistributedDataParallel` and moving to correct device.
+            train.torch.prepare_model(...)
+
+            # Configures the dataloader for distributed training by adding a
+            # `DistributedSampler`.
+            # You should NOT use this if you are doing
+            # `train.get_dataset_shard(...).to_torch(...)`
+            train.torch.prepare_data_loader(...)
+
+            # Returns the current torch device.
+            train.torch.get_device()
+
+    To save a model to use for the ``TorchPredictor``, you must save it under the
+    "model" kwarg in ``train.save_checkpoint()``.
+
+    Example:
+        .. code-block:: python
+
+            import torch
+            import torch.nn as nn
+
+            import ray
+            from ray import train
+            from ray.ml.integrations.torch import TorchTrainer
+
+            input_size = 1
+            layer_size = 15
+            output_size = 1
+
+            class NeuralNetwork(nn.Module):
+                def __init__(self):
+                    super(NeuralNetwork, self).__init__()
+                    self.layer1 = nn.Linear(input_size, layer_size)
+                    self.relu = nn.ReLU()
+                    self.layer2 = nn.Linear(layer_size, output_size)
+
+                def forward(self, input):
+                    return self.layer2(self.relu(self.layer1(input)))
+
+            def train_loop_for_worker():
+                dataset_shard = train.get_dataset_shard("train")
```

Comment:
```suggestion
                dataset_shard = train.get_train_dataset_shard()
```

Reply:
This should contain "train" key -> we decided to make this explicitly required.